jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -246,22 +248,21 @@ private SpanContext createNewContext(String debugID) {
       return new SpanContext(id, id, 0, flags);
     }
 
-    private SpanContext createChildContext() {
+    private SpanContext createChildContext(SpanContext preferredParent) {
       // For server-side RPC spans we reuse spanID per Zipkin convention
       if (isRPCServer()) {
-        if (parent.isSampled()) {
+        if (preferredParent.isSampled()) {
           metrics.tracesJoinedSampled.inc(1);
         } else {
           metrics.tracesJoinedNotSampled.inc(1);
         }
-        return parent;
```

Comment:
App is instrumented correctly, the problem was that I was trying this PR when `SPAN_KIND` tags were not added due to zipkin sr,cs.. annotations. Now it is working fine.